hasjob/models/domain.py

```diff
@@ -52,6 +53,15 @@ def use_title(self):
     def has_profile(self):
         return bool(self.title and self.description)
 
+    @cached_property
+    def get_logo_url(self):
+        """Returns logo_url if present, else returns the logo from its most recent job post"""
+        if self.logo_url:
+            return self.logo_url
+        else:
+            post = self.jobposts.filter(JobPost.company_logo != None).order_by("datetime desc").first()  # NOQA
```

Comment:
Filter by `POSTSTATUS.ARCHIVED` since you don't want the logo from something marked as spam, etc.